packages/api/tests/app/test-auth.js

```diff
@@ -0,0 +1,192 @@
+  t.context.ensureAuthorizedApp.use(boom());
+  t.context.ensureAuthorizedApp.get(
+    '/asdf',
+    ensureAuthorized,
+    (_req, res) => res.status(200).json({ name: 'John Gillespie Magee Jr.' })
+  );
+});
+
+test.after.always(async (t) => {
+  const { accessTokenModel, userModel } = t.context;
+
+  await accessTokenModel.deleteTable();
+  await userModel.deleteTable();
+});
+
+test.serial('ensureAuthorized() with a valid oauth jwtToken succeeds', async (t) => {
+  const { accessTokenModel, ensureAuthorizedApp, userModel } = t.context;
+
+  const accessToken = fakeAccessTokenFactory();
+  await accessTokenModel.create(accessToken);
+
+  const jwtToken = createJwtToken(accessToken);
+
+  await userModel.create({ userName: accessToken.username });
+
+  process.env.OAUTH_PROVIDER = 'earthdata';
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', `Bearer ${jwtToken}`)
+    .then((res) => {
+      t.is(res.statusCode, 200);
+    });
+});
+
+test('ensureAuthorized() returns an unauthorized response if an authorization header is not set', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'Authorization header missing');
+    });
+});
+
+test('ensureAuthorized() returns an unauthorized response if an authorization scheme is not "Bearer"', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', 'Basic asdf')
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'Authorization scheme must be Bearer');
+    });
+});
+
+test('ensureAuthorized() returns an unauthorized response if the token is missing', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', 'Bearer')
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'Missing token');
+    });
+});
+
+test.serial('ensureAuthorized() verifies that the user is authorized when not using Launchpad authentication', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  process.env.OAUTH_PROVIDER = 'earthdata';
+
+  const jwtToken = createJwtToken();
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', `Bearer ${jwtToken}`)
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'User not authorized');
+    });
+});
+
+test.serial('If Launchpad authentication is enabled, and the Bearer token is not a valid JWT token, ensureAuthorized() passes the authentication over to ensureLaunchpadAPIAuthorized()', async (t) => {
```

Comment:
```suggestion
test.serial('If Launchpad authentication is enabled, and the Bearer token is not a JWT token, ensureAuthorized() passes the authentication over to ensureLaunchpadAPIAuthorized()', async (t) => {
```

Sort of semantics, but the handover happens when the JWT verification fails because it is "malformed", meaning it's not really a JWT. JWT validity is more related to the secret the token was signed with, its algorithm, etc